examples/gitserver/README.md

```diff
@@ -1,16 +1,220 @@
+
+    ```sh
+    $ oc env dc/gitserver PUBLIC_URL=http://gitserver-myproject.infra.openshift.com
+    ```
+
+    This will result in a new deployment of the Git Server with the updated environment 
+    variable.
+
+
+### Push code to the Git Server
+
+Code repositories can be initialized on the Git Server by either doing a 'git push' or
+a 'git clone' of the repository.
+
+1. Setup your credentials
+
+   By default the Git Server will allow users or service accounts that can create pods in
+   the project namespace to create and push code to the Git Server. The easiest way to 
+   provide credentials to the Git Server is by using a custom credential helper that will 
+   send your OpenShift token by default to the server.
+
+   ```sh
+   $ git config --global credential.http://gitserver-myproject.infra.openshift.com.helper \
+         '!f() { echo "username=$(oc whoami)"; echo "password=$(oc whoami -t)"; }; f'
+   ```
+
+   **NOTE:** the config key is `credential.[your server URL here].helper`
+
+2. Clone a new repository
+   
+   ```
+   $ git clone http://gitserver-myproject.infra.openshift.com/ruby-hello-world.git
+   ```
+
+   **NOTE:** the ruby-hello-world.git repository does not exist before running this command. 
+   By cloning it, you are creating it in the Git Server.
+
+3. Populate the repository with code
+
+   Add a public Github repository as a remote, fetch it, and push its code to your
+   local Git Server.
+
+   ```sh
+   $ cd ruby-helllo-world
+   $ git remote add github https://github.com/openshift/ruby-hello-world.git
+   $ git fetch github
+   $ git rebase github/master
+   $ git push origin master
+   ```
+
+### Build and Deploy code from the Git Server
+
+1. Create artifacts using `oc new-app`
+
+   ```sh
+   $ oc new-app http://gitserver-myproject.infra.openshift.com/ruby-hello-world.git
+   ```
+
+2. Setup a source secret for the BuildConfig
+
+   In order for an Openshift build to be able to clone code from the gitserver, it must be
+   able to provide credentials to it. The secret for the gitserver service account can be
+   used for this purpose.
+
+   To get the name of a secret, run:
+   ```sh
+   $ oc get secrets | grep gitserver | grep token | head -n 1
+   ```
+
+   Edit the BuildConfig and set the source secret using the secret name returned above.
+   ```sh
+   $ oc edit bc ruby-hello-world
+   ```
+   For an example, see Step 3 here: https://docs.openshift.org/latest/dev_guide/builds.html#basic-authentication
+
+3. Start a new build
+
+   A build can be started explicitly
+
+   ```sh
+   $ oc start-build ruby-hello-world
+   ```
+
+   Or by pushing new commits to your Git Server
+   ```sh
+   $ git commit --amend
+   $ git push origin master
+   ```
+
+### Secure the Git Server
+
+1. If your cluster's router has been setup with a default certificate for edge termination,
+   just modify the route to use edge termination:
+
+   ```sh
+   $ oc edit route gitserver
```

Comment:
Right, we are not going to enhance `oc expose` for creating secured routes since the command is already bloated from upstream, that's why we spinned off `create route`.